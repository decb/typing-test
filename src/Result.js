import React from "react";
import { Alert } from "antd";

import { wordCalculation, charCalculation } from "./resultCalculator";
import { buildMessage } from "./resultTextBuilder";

function Result(props) {
  const { inputWords, targetWords } = props;
  const wordsTyped = inputWords.length;
  const [wordsCorrect, wordsCorrectPercent] = wordCalculation(
    wordsTyped,
    inputWords,
    targetWords
  );

  const charsTyped = inputWords.join("").split("").length;
  const [charsCorrect, charsCorrectPercent] = charCalculation(
    charsTyped,
    inputWords,
    targetWords
  );

  const finalMessage = buildMessage(
    wordsTyped,
    wordsCorrect,
    wordsCorrectPercent,
    charsTyped,
    charsCorrect,
    charsCorrectPercent
  );

  return (
    <Alert message="Test complete" description={finalMessage} type="success" />
  );
}

export default Result;
