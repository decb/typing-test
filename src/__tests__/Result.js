import React from "react";
import renderer from "react-test-renderer";

import Result from "../Result";

const inputWords = ["a"];
const targetWords = ["b"];

describe("<Result />", () => {
  test("Renders correctly", () => {
    const tree = renderer
      .create(<Result inputWords={inputWords} targetWords={targetWords} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
