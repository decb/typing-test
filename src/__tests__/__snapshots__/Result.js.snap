// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Result /> Renders correctly 1`] = `
<div
  className="ant-alert ant-alert-success ant-alert-with-description ant-alert-no-icon"
  data-show={true}
>
  <span
    className="ant-alert-message"
  >
    Test complete
  </span>
  <span
    className="ant-alert-description"
  >
    In one minute, you typed 1 words, of which 0 were correct (0%). This amounted to 1 characters, of which 0 were correct (0%). To try again, press the reset button or select a new text source.
  </span>
</div>
`;
